Tribler/Test/API/test_download.py

```diff
@@ -33,11 +36,20 @@ def setUpPostSession(self):
         pass
 
     def test_download_torrent_from_url(self):
-        d = self.session.start_download_from_uri(TORRENT_R)
+        # Setup file server to serve torrent file
+        files_path = os.path.join(self.session_base_dir, 'http_torrent_files')
+        os.mkdir(files_path)
+        shutil.copyfile(TORRENT_FILE, os.path.join(files_path, 'ubuntu.torrent'))
+        file_server_port = get_random_port()
+        self.setUpFileServer(file_server_port, files_path)
+
+        d = self.session.start_download_from_uri('http://localhost:%s/ubuntu.torrent' % file_server_port)
         self._logger.debug("Download started: %s", d)
         d.set_state_callback(self.downloader_state_callback)
         assert self.downloading_event.wait(60)
 
+        self.file_server.stopListening()
```

Comment:
You should be using setUp/tearDown for this kind of stuff. If the test fails, you will leak the file_server.